packages/core/src/html/NodeProcessor.js

```diff
@@ -190,10 +191,9 @@ class NodeProcessor {
         this.footnoteProcessor.processMbTempFootnotes(node);
         break;
       case 'script':
-        processScriptTag(node);
-        break;
       case 'style':
-        processStyleTag(node);
+        processScriptAndStyleTag(node, this.docId === 'layout'
+          ? this.config.layoutUserScriptsAndStyles : this.config.pageUserScriptsAndStyles);
```

Comment:
how about passing a bucket `[]` from `Layout/Page`? So this `NodeProcessor` dosen't have to be aware of how it's being used

Reply:
(or maintain the bucket here and pull it out in `Layout/Page`)

Reply:
Thanks for the suggestion! This way is much cleaner, not sure why it didn't occur to me earlier 😕